l3_dump.py

```diff
@@ -237,7 +254,15 @@ def main():
 
     strings = parse_rodata_string_offsets(program_bin)
 
-    with open(sys.argv[1], 'rb') as file:
+    # To enable unit-testing, via pytests, the parsing and return-data logic
```

Comment:
Are you saying that returning a bunch of lists, as opposed to a list of tuples, somehow makes it easier to unit test? It seems slightly odd to this. i.e. maintaining a list of tuples seems cleaner than a bunch of parallel lists.

Reply:
I, too, thought of returning a list of tuples when writing this first. But that got more difficult to work with for unit-testing.

As an example, see `test_unit_test_dump_log_entries()` in `pytests/l3_dump_test.py`.

One of the validations we do there is to create a hand-rolled expected-messages list and compare with the list output by this `do_main()` method.

```
 88     # Invoke the L3-dump script to unpack the slow-log-entries.
 89     (nentries, tid_list, loc_list, msg_list, arg1_list, arg2_list) \
 90         = l3_dump.do_main(['/tmp/l3.c-small-unit-test.dat', binary],
 91                           return_logentry_lists = True)
 92
 93     # -----------------------------------------------------------------------
 94     # Here is where it gets interesting. Hard-coded verification of expected
 95     # message and arguments, as unpacked by utility. This is one example of
 96     # verification using hard-coded expected-outputs lists.
 97     # -----------------------------------------------------------------------
 98     exp_msg_list = [  'Simple-log-msg-Args(1,2)'
 99                     , 'Simple-log-msg-Args(3,4)'
100                     , 'Potential memory overwrite (addr, size)'
101                     , 'Invalid buffer handle (addr)'
102                    ]
103     assert msg_list == exp_msg_list
```

It became easy to do L103; i.e., simply compare actual and expected list of messages.

If `do_main()` were to return a list of tuples, I would have to crack that open in this verifier to create the list of actual messages, i.e., `msg_list`.

I don't see the current method too too bad. Let me get this in, sit on it, and if I see benefits to returning a list of tuples, I can change all this easily enough.